### PR TITLE
feat: add logo verification logic and user state persistence

### PR DESCRIPTION
**Pull request by caroso1222**
This PR contains the following things:
1. Validates the logo guess
2. Persists the validations into the user state (now the users know what logos they have guessed)

Note: I know some of the things in the PR might look hacky, but I just wanted to ship something functional (done better than perfect). We can improve along the way.

### CODE REVIEW

File: apps/api/src/app/logo/logo.controller.ts [hunk 1]
```diff
@@ -16,14 +21,39 @@ export class LogoController {
...
+  async findById(@Param('id') id: string, @Req() request: Request): Promise<Logo> {
+    const user = request['user'];
     const logo = await this.logoService.findOne(id);
     const logoPayload = logo.toJSON() as Logo;
     let obfuscatedName = logoPayload.name.replace(/\w/gi, '*');
     obfuscatedName = obfuscatedName.replace(/ /g, '_');
     logoPayload.obfuscatedName = obfuscatedName;
-    delete logoPayload.realImageUrl;
+    const validated = await this.userStateService.verifyValidatedLogo(id, user.id);
+    console.log(validated);
```
Comment (edgargonzalez525) [inline apps/api/src/app/logo/logo.controller.ts:52]: should we keep this here?

File: apps/api/src/shared/service/user-state.service.ts [hunk 2]
```diff
@@ -20,8 +21,25 @@ export class UserStateService {
   async findByUser(userId: string): Promise<UserState> {
     const state: UserState = await this.userStateModel.findOne({ user: userId }).exec();
     if (state) {
-      state.logos = await this.userCompletedLogoService.findByState(state._id);
+      // state.logos = await this.userCompletedLogoService.findByState(state._id);
```
Comment (edgargonzalez525) [inline apps/api/src/shared/service/user-state.service.ts:24]: should this be removed along the condition?

File: apps/api/src/shared/service/user-state.service.ts [hunk 3]
```diff
@@ -20,8 +21,25 @@ export class UserStateService {
...
     }
     return state;
   }
+
+  async verifyValidatedLogo(logoId: string, userId: string): Promise<boolean> {
+    const logos = await this.getUserLogos(userId);
+    console.log(logos);
```
Comment (edgargonzalez525) [inline apps/api/src/shared/service/user-state.service.ts:31]: do we need this?

File: apps/api/src/shared/service/user.service.ts [hunk 4]
```diff
@@ -26,35 +28,41 @@ export class UserService {
     return user;
   }
 
-  async login(credentials: { email: string, password: string }) {
+  async login(credentials: { email: string; password: string }) {
+    const users = await this.userModel.find();
+    console.log(users);
```
Comment (edgargonzalez525) [inline apps/api/src/shared/service/user.service.ts:33]: do we need this?

File: apps/api/src/shared/service/user.service.ts [hunk 5]
```diff
@@ -26,35 +28,41 @@ export class UserService {
...
     return await instance.save();
   }
 
   async getLevelLogos(userId: string, levelId: string): Promise<UserLogo[]> {
     const user = await this.findOne(userId);
     const level = await this.levelService.findOne(levelId);
     return await level.logos.map((logo: Logo) => {
-      const completed = user.state.logos
-      .findIndex((state) => state.logo.toString() === logo._id.toString()) !== -1;
+      console.log(user.state.logos);
```
Comment (edgargonzalez525) [inline apps/api/src/shared/service/user.service.ts:63]: do we need this?

File: apps/logo-quiz/src/app/views/LogoVerify/LogoVerify.tsx [hunk 6]
```diff
@@ -27,8 +30,11 @@ interface LogoVerifyProps extends RouteComponentProps<MatchParams> {
   removeLetterFromGuess: typeof removeLetterFromGuess;
   fetchLogo: typeof fetchLogo;
   flushLogo: typeof flushLogo;
-  guess: GuessedLetter[];
+  validateLogo: typeof validateLogo;
```
Comment (edgargonzalez525) [inline apps/logo-quiz/src/app/views/LogoVerify/LogoVerify.tsx:33]: what is this typeof doing here?

File: apps/logo-quiz/src/shared/services/level.service.ts [hunk 7]
```diff
@@ -6,8 +6,15 @@ import { authHeader } from '../helpers';
...
-    .then((level: AxiosResponse<Level[]>) => {
-      return level.data;
-    });
+  }).then((levels: AxiosResponse<Level[]>) => {
+    return levels.data;
+  });
+}
+
+export function fetchLevel(id: String) {
+  return axios.get(`http://localhost:3333/api/levels/${id}`, {
```
Comment (edgargonzalez525) [inline apps/logo-quiz/src/shared/services/level.service.ts:15]: should we move this localhost to a configuration variable?

Comment (edgargonzalez525) [reply apps/logo-quiz/src/shared/services/level.service.ts:15]: same in other places

File: apps/logo-quiz/src/store/levels/types.ts [hunk 8]
```diff
@@ -1,12 +1,12 @@
-import { Level } from '@logo-quiz/models';
-import { AxiosError } from 'axios';
+import { Level } from "@logo-quiz/models";
```
Comment (edgargonzalez525) [inline apps/logo-quiz/src/store/levels/types.ts:1]: why using double quotes?

File: apps/logo-quiz/src/store/levels/types.ts [hunk 9]
```diff
@@ -1,12 +1,12 @@
...
+import { AxiosError } from "axios";
 
-export const REQUEST_LEVELS = 'REQUEST_LEVELS';
-export const REQUEST_LEVELS_SUCCESS = 'REQUEST_LEVELS_SUCCESS';
-export const REQUEST_LEVELS_ERROR = 'REQUEST_LEVELS_ERROR';
+export const REQUEST_LEVELS = "REQUEST_LEVELS";
```
Comment (edgargonzalez525) [inline apps/logo-quiz/src/store/levels/types.ts:4]: same here

File: apps/logo-quiz/src/store/levels/types.ts (hunk 8 above)
Comment (caroso1222) [reply apps/logo-quiz/src/store/levels/types.ts:1]: Sorry I had prettier enabled with the wrong config. Will fix it.

File: apps/logo-quiz/src/shared/services/level.service.ts (hunk 7 above)
Comment (caroso1222) [reply apps/logo-quiz/src/shared/services/level.service.ts:15]: Yeah, sounds good, I'll do it

File: apps/logo-quiz/src/app/views/LogoVerify/LogoVerify.tsx (hunk 6 above)
Comment (caroso1222) [reply apps/logo-quiz/src/app/views/LogoVerify/LogoVerify.tsx:33]: We need a way to tell the component "the function 'validateLogo' is valid, you can use it". We do it by creating this interface `LogoVerifyProps`. As `validateLogo` is a function itself, we cannot use it as a type like `validateLogo: validateLogo`. Therefore, we need to tell the component "you can use `validateLogo` which is `typeof validateLogo`".

I know it's super confusing like that. Read this for full explanation.

File: apps/api/src/shared/service/user-state.service.ts (hunk 3 above)
Comment (caroso1222) [reply apps/api/src/shared/service/user-state.service.ts:31]: Remoted all console.logs, thanks for noticing

File: apps/api/src/shared/service/user-state.service.ts (hunk 2 above)
Comment (caroso1222) [reply apps/api/src/shared/service/user-state.service.ts:24]: yep, removed

**Pull request merged**